src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -297,6 +347,112 @@ export default {
+        if (selectableItems.length) {
+          [nextItem] = selectableItems;
+        }
+      }
+      this.activateItemViaKeyboardNavigation(nextItem);
+    },
+    onArrowUpKeyDown() {
+      if (!this.isMultiple || this.isOpen) {
+        this.onArrowKey(ARROW_UP);
+      }
+    },
+    onArrowDownKeyDown() {
+      if (!this.isMultiple || this.isOpen) {
+        this.onArrowKey(ARROW_DOWN);
+      }
+    },
+    onEnterOrSpaceKeyDown() {
+      if (this.isOpen) {
+        if (!this.isMultiple || !this.currentActiveItem) {
+          this.closeViaKeyboardNavigation();
+        } else {
+          const keyboardNavigation = true;
+          this.select(this.currentActiveItem, keyboardNavigation);
+        }
+      } else {
+        this.show();
+      }
+    },
+    onSearchFieldEnterKeyDown() {
+      this.closeViaKeyboardNavigation();
+    },
+    onTabKeyDown() {
+      this.closeViaKeyboardNavigation();
+    },
+    activateItemViaKeyboardNavigation(item) {
+      if (item) {
+        if (this.isMultiple) {
+          this.active = item;
+          this.$refs.popover.update();
+        } else {
+          const keyboardNavigation = true;
+          this.select(item, keyboardNavigation);
+        }
+      }
+    },
+    closeViaKeyboardNavigation() {
+      if (this.isOpen) {
+        this.hide();
+        if (this.isSearchEnabled) {
+          // if the search is active the focus is lost from the trigger, then it must regain the focus
+          const $el = this.$refs.popover.$el.querySelector(this.initialFocusedElementType);
+          if ($el) {
+            $el.focus();
+          }
+          this.initialFocusedElementType = null;
+        }
+      }
+    },
+    updateScroll() {
+      const currentItemIndex = this.filteredItems.indexOf(this.currentActiveItem);
+      const $elItems = this.$refs.itemElements;
+
+      if ($elItems && $elItems.length && currentItemIndex >= 0) {
+        const itemTop = $elItems[currentItemIndex].offsetTop;
+        const itemBottom = itemTop + $elItems[currentItemIndex].offsetHeight;
+        const visibleWindowHeight = this.$refs.itemsOverflowContainer.clientHeight;
+
+        if (!this.visibleWindow) {
+          // Initial visibleWindow setup
+          this.visibleWindow = { top: 0, bottom: visibleWindowHeight };
+        }
+
+        if (itemBottom >= this.visibleWindow.bottom) {
+          this.visibleWindow.top = itemBottom - visibleWindowHeight;
+          this.visibleWindow.bottom = itemBottom;
+        } else if (this.isFirstItemCustom && currentItemIndex === 0) {
+          this.visibleWindow.top = 0;
+          this.visibleWindow.bottom = visibleWindowHeight;
+        } else if (itemTop <= this.visibleWindow.top) {
+          this.visibleWindow.top = itemTop;
+          this.visibleWindow.bottom = this.visibleWindow.top + visibleWindowHeight;
+        }
+        this.$refs.itemsOverflowContainer.scrollTop = this.visibleWindow.top;
+      }
+    },
+    removeActiveItem() {
```

Comment:
`clearLastHighligthedItem` would be a better name. Remove indicates it is going to be removed from the list and user cannot select it anymore.